### PR TITLE
Add Embree 4.3.0

### PR DESCRIPTION
**Pull request by Vertexwahn**
Add Embree 4.3.0

### CODE REVIEW

**Comment by Vertexwahn** [comment]
No release archive URL available - please add skip-url-stability-check tag

File: modules/embree/4.3.0/MODULE.bazel [hunk 1]
```diff
@@ -0,0 +1,8 @@
+module(
+  name = "embree", 
+  version = "4.3.0"
+)
+
+bazel_dep(name = "bazel_skylib", version = "1.4.2")
+bazel_dep(name = "onetbb", version = "2021.10.0")
+bazel_dep(name = "rules_cc", version = "0.0.9")
```
Comment (fmeum) [inline modules/embree/4.3.0/MODULE.bazel:8]: Needs a dep on `platforms` (see CI failures)

**Comment by Vertexwahn** [comment]
@fmeum Please review again. There was a linker error. Fixed now. BTW: Here is a test rendering using Embree 4.3.0: 
- test

File: modules/embree/4.3.0/patches/bazel_build.patch [hunk 2]
```diff
@@ -0,0 +1,658 @@
...
++config_setting(
++    name = "embree_internal_tasking_system",
++    define_values = {"tasking_system": "internal"},
++)
++
++COMMON_DEFINES = select({
++    ":embree_internal_tasking_system": [
++        "TASKING_INTERNAL",
++        "EMBREE_TARGET_SSE2",
++        #"EMBREE_TARGET_SSE42",
```
Comment (fmeum) [inline modules/embree/4.3.0/patches/bazel_build.patch:33]: Could use a comment explaining why these are commented out

File: modules/embree/4.3.0/patches/bazel_build.patch [hunk 3]
```diff
@@ -0,0 +1,658 @@
...
++config_setting(
++    name = "windows_x86_64",
++    values = {"cpu": "x64_windows"},
++    visibility = ["//visibility:public"],
++)
++
++load("@bazel_skylib//rules:expand_template.bzl", "expand_template")
++
++# If you want to use the internal tasking system use:
++# bazel build //... --define tasking_system=internal
```
Comment (fmeum) [inline modules/embree/4.3.0/patches/bazel_build.patch:22]: Could we namespace this define with `embree`? Alternatively, you could use a custom Starlark flag from skylib.

File: modules/embree/4.3.0/patches/bazel_build.patch [hunk 4]
```diff
@@ -0,0 +1,658 @@
...
++                "-ldl",
++                "-lpthread",
++                "-lrt",
++            ],
++            "//conditions:default": [
++                "-lpthread",
++            ],
++        }),
++    #textual_hdrs = [
++    #    ":config_h",
```
Comment (fmeum) [inline modules/embree/4.3.0/patches/bazel_build.patch:647]: See comment about comments :⁠-⁠)

File: modules/embree/4.3.0/patches/bazel_build.patch [hunk 5]
```diff
@@ -0,0 +1,658 @@
...
++    SPDX-FileCopyrightText: 2023 Julian Amann <dev@vertexwahn.de>
++    SPDX-License-Identifier: Apache-2.0
++"""
++
++package(
++    default_visibility = ["//visibility:public"],
++)
++
++config_setting(
++    name = "windows_x86_64",
```
Comment (fmeum) [inline modules/embree/4.3.0/patches/bazel_build.patch:14]: This won't work with toolchain resolution, which is enabled by default in Bazel 7.

Comment (Vertexwahn) [reply modules/embree/4.3.0/patches/bazel_build.patch:14]: Does a select with `"@platforms//os:windows"` work here, i.e. replacing

```
+}) + select({
+    ":windows_x86_64": [
+        "__SSE__",
+        "__SSE2__",
+        "EMBREE_STATIC_LIB",
+    ],
+    "//conditions:default": [
+    ],
+})
```

by

```
+}) + select({
+    "@platforms//os:windows": [
+        "__SSE__",
+        "__SSE2__",
+        "EMBREE_STATIC_LIB",
+    ],
+    "//conditions:default": [
+    ],
+})
```

?

File: modules/embree/4.3.0/patches/bazel_build.patch (hunk 2 above)
Comment (Vertexwahn) [reply modules/embree/4.3.0/patches/bazel_build.patch]: Added a comment

File: modules/embree/4.3.0/patches/bazel_build.patch (hunk 3 above)
Comment (Vertexwahn) [reply modules/embree/4.3.0/patches/bazel_build.patch]: Added embree_ prefix

File: modules/embree/4.3.0/patches/bazel_build.patch (hunk 5 above)
Comment (Vertexwahn) [reply modules/embree/4.3.0/patches/bazel_build.patch]: Switched to @platforms//os:windows

**Comment by Vertexwahn** [comment]
@fmeum Please review again. Again a test rendering using Embree 4.3.0:

specular

**Review by fmeum — APPROVED** [review]
Thanks for the renders, that's a nice "tip" for a review :⁠-⁠)

**Pull request merged**